src/nat/llm/huggingface_llm.py

```
@@ -0,0 +1,289 @@
+        builder: The NAT builder instance.
+
+    Yields:
+        LLMProviderInfo: Provider information for the loaded model.
+    """
+    try:
+        import torch
+        from transformers import AutoModelForCausalLM
+        from transformers import AutoTokenizer
+    except ImportError as err:
+        raise ImportError(
+            "transformers and torch required. Install: pip install transformers torch accelerate") from err
```

Ruff (0.14.8)

Avoid specifying long messages outside the exception class

(TRY003)